geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSPopIntegrationTest.java

```diff
@@ -52,213 +56,137 @@ public void tearDown() {
   }
 
   @Test
-  public void givenKeyNotProvided_returnsWrongNumberOfArgumentsError() {
-    assertThatThrownBy(() -> jedis.sendCommand("key", Protocol.Command.SPOP))
-        .hasMessageContaining("ERR wrong number of arguments for 'spop' command");
+  public void spopTooFewArgs_returnsError() {
+    assertAtLeastNArgs(jedis, Protocol.Command.SPOP, 1);
   }
 
   @Test
-  public void givenMoreThanThreeArguments_returnsSyntaxError() {
+  public void spopTooManyArgs_returnsError() {
     assertThatThrownBy(
-        () -> jedis.sendCommand("key", Protocol.Command.SPOP, "key", "NaN", "extraArg"))
+        () -> jedis.sendCommand(SET_KEY, Protocol.Command.SPOP, SET_KEY, "5", "5"))
             .hasMessageContaining(ERROR_SYNTAX);
```

Comment:
This test can be simplified by using the `RedisCommandArgumentsTestHelper.assertAtMostNArgs()` method.

Reply:
By just calling that method or assertThatThrownBy(() ->RedisCommandArgumentsTestHelper.assertAtMostNArgs()).hasMessageContaining(ERROR_SYNTAX);

When I was using RedisCommandArgumentsTestHelper.assertAtMostNArgs() on it's own it failed both the native and integration test.

Reply:
Aah, I see what the problem is. When SPOP has too few arguments it uses the "ERR wrong number of arguments for 'spop' command" error message, but when it has too many arguments, it uses the "ERR syntax error" error message. The `assertAtMostNArgs()` method only expects to see the first error message, which is why it was failing.

I withdraw this comment, the test was right to begin with.